### DISCUSSION TITLE
bitrateStrategy supported for SRT?

### DISCUSSION THREAD

**Discussion by mkrn** [HaishinKit/HaishinKit.swift#1598] [Q&A]
Hello, Im curious if adaptive bitrate can work for SRT?
Or what are the default options for SRT, does it apply any sort of error recovery by default? 
What's the best way to detect if signal is bad and bitrate should be reduced?
Thanks

**Comment by shogo4405** [comment]
## 1.9.x
Unfortunately, it's not implemented in the 1.9.x series. However, similar streaming statistics can be obtained as follows: https://github.com/shogo4405/HaishinKit.swift/blob/dbeec845e62f01808bcdffe00ee96a5e9eef5792/SRTHaishinKit/SRTConnection.swift#L31
Details can be found in the following header information: https://github.com/Haivision/srt/blob/master/srtcore/srt.h

## 2.0.0
It has been implemented in the 2.0.x series. Both RTMPStream and SRTStream now have the HKStream protocol, which can also be used with SRT. https://github.com/shogo4405/HaishinKit.swift/wiki/Moving-from-1.9.x-APIs-to-2.0.0#200-3

**Comment by mkrn** [comment]
Thanks, i've figured this much..
Is the FPS consistent in SRT? Struggling to make it work with my backend server..
![image](https://github.com/user-attachments/assets/6501ba6d-742e-47d1-aa73-1c180c1876a9)

getting severe issues even when only a few packets dropped in performance data::
```
Performance Data:
- Total Packets Sent: 34668 
- Total Send Packet Loss: 3 
- Total Retransmitted Packets: 2
- Total Received ACKs: 10888
- Total Received NAKs: 3
- Packets In Flight: 8
- Total Bytes Sent: 33419216 
- Sending Bitrate (Mbps): 1.6901517447747045
- Total Dropped Sent Packets: 0  
- Round Trip Time (ms): 68.579
- Congestion Window: 8192
- Packets in Send Buffer: 12
- Bytes in Send Buffer: 11754
- Packet Congestion Window: 8192
- Bytes Available in Send Buffer: 12270000
- pktReorderTolerance: 0
- pktSndFilterExtra: 0
```
How are you testing SRT?

**Reply by shogo4405** [reply]
Personally, I am not using SRT, so the testing is not sufficient. The tests are mainly conducted in the following environments:
1. Using ffplay on localhost:
   - ffplay -fflags nobuffer 'srt://0.0.0.0:9998?mode=listener'
   - ffmpeg -stream_loop -1 -re -i filename.mp4 -c copy -f mpegts 'srt://0.0.0.0:9998?mode=listener'
1. On localhost as follows:
    - SRS: https://github.com/ossrs/srs
1. Sending and receiving data between HaishinKit on the same network
1. External server as follows:
   - Cloudflare
